src/WebApp/Components/Chatbot/Chatbot.razor

```diff
@@ -53,10 +53,11 @@
 
     protected override async Task OnInitializedAsync()
     {
-        if (ChatConfig.TryReadFromConfig(Configuration, out var chatConfig))
+        var kernel = ServiceProvider.GetService<Kernel>();
```

Comment:
@SteveSandersonMS @javiercn - this was the way I figured out how to make an "optional" DI service work in Blazor. See https://github.com/dotnet/razor/issues/7653.

Is there a better way?

Reply:
> Is there a better way?

Nope, currently this is the way.

Reply:
If the `GetService` call bothers you, another fairly idiomatic option would be having some helper that gets the kernel and returns null or throws if the required config is absent. That could be an extension method on `IServiceProvider`. But what you have here is completely reasonable already.

Reply:
> If the GetService call bothers you

It's mostly injecting `IServiceProvider` that bothers me. It's not a huge concern, but it is usually frowned upon - https://stackoverflow.com/questions/70264338/inject-iserviceprovider-into-an-api-controller-or-individual-services.

> what you have here is completely reasonable already.

Thanks for verifying.